dotCMS/src/main/java/com/dotcms/rest/api/v1/index/ESIndexResource.java

```diff
@@ -109,6 +110,7 @@ protected InitDataObject auth(String params,HttpServletRequest request) throws D
         return init;
     }
 
+    @CloseDBIfOpened
     public static void restoreIndex(final File file, final String alias, String index, final boolean clear) throws DotDataException {
```

Comment:
2 things,
The thread could be triggered by the dotSubmmiter.
You can encapsulate the logic into the Thread in order to wrapInTransaction it

Reply:
yeah put annotation at `Thread.run` method level instead and remove the `finally` block